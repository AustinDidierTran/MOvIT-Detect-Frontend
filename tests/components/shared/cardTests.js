/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import React from 'react';
import toJson from 'enzyme-to-json';
import CustomCard from '../../../src/components/shared/card';

Enzyme.configure({ adapter: new Adapter() });

describe('CustomCard Tests', () => {
  const props = {
    element: <div>Test</div>,
    header: <span>This is a test</span>,
  };

  it('should have proptypes', () => {
    const actualValue = CustomCard.propTypes;

    const expectedValue = {
      element: PropTypes.element.isRequired,
      header: PropTypes.element.isRequired,
      style: PropTypes.object,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<CustomCard {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
