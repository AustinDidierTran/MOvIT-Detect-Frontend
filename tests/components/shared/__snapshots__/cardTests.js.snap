// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCard Tests should match the snapshot 1`] = `
<div
  className="container"
>
  <Card
    className="card"
    footer={null}
    header={
      <span>
        This is a test
      </span>
    }
    id={null}
    style={null}
    subtitle={null}
    title={null}
  >
    <div>
      Test
    </div>
  </Card>
</div>
`;
